packages/flutter_tools/lib/src/asset.dart

```diff
@@ -395,28 +395,23 @@ List<_Asset> _getMaterialAssets(String fontSet) {
 }
 
 
-/// Processes dependencies into a string representing the license file.
+/// Processes dependencies into a string representing the NOTICES file.
 ///
-/// Reads the LICENSE file from each package in the .packages file, splitting
-/// each one into each component license (so that we can de-dupe if possible).
-/// If provided with a pubspec.yaml file, only direct depedencies are included
-/// in the resulting LICENSE file.
+/// Reads the NOTICES or LICENSE file from each package in the .packages file,
+/// splitting each one into each component license so that it can be de-duped
+/// if possible. If the NOTICES file exists, it is preferred over the LICENSE
+/// file.
 ///
 /// Individual licenses inside each LICENSE file should be separated by 80
 /// hyphens on their own on a line.
 ///
-/// If a LICENSE file contains more than one component license, then each
-/// component license must start with the names of the packages to which the
-/// component license applies, with each package name on its own line, and the
-/// list of package names separated from the actual license text by a blank
-/// line. (The packages need not match the names of the pub package. For
+/// If a LICENSE or NOTICES file contains more than one component license,
+/// then each component license must start with the names of the packages to
+/// which the component license applies, with each package name on its own line
+/// and the list of package names separated from the actual license text by a
+/// blank line. The packages need not match the names of the pub package. For
 /// example, a package might itself contain code from multiple third-party
-/// sources, and might need to include a license for each one.)
-// Note: this logic currently has a bug, in that we collect LICENSE information
-// for dev_dependencies and transitive dev_dependencies. These are not actually
-// compiled into the released application and don't need to be included. Unfortunately,
-// the pubspec.yaml alone does not have enough information to determine which
-// dependencies are transitive of dev_dependencies, so a simple filter isn't sufficient.
```

Comment:
was this note wrong or did it get fixed?

Reply:
It was wrong, per our discussion a while back :)

Reply:
I guess we talked about it like, months ago, but seems like it was last week!